R/axis_handling.R

```diff
@@ -0,0 +1,151 @@
+#'labeling functions in the \code{scales::label_*} family.
+#'
+#'@importFrom stringr str_length
+#'
+#'@examples
+#'\dontrun{
+#' # Default implementation - this will abbreviate all labels except the first
+#' # for both continuous and date scales, using their respective abbreviation
+#' # function. Note the syntax - this function actually produces another
+#' # function, and so you must include the opening and closing parentheses
+#' # following the function's name (i.e., \code{abbr_years_cont()}).
+#'
+#' df2 <- dplyr::mutate(transit_ridership,year2 = as.Date(paste0(year,"-01-01")))
+#' df1 <- filter(df2,year >= 2000)
+#'
+#' ggplot(df1,
+#'        aes(x = year, y = ridership, color = system)) +
+#'   geom_line() +
+#'   scale_x_continuous(labels = abbr_years_cont())
+#'
+#' ggplot(df1,
+#'        aes(x = year2, y = ridership, color = system)) +
+#'   geom_line() +
+#'   scale_x_date(labels = abbr_years_date())
+#'
+#' # If customizations are desired, users can use \code{full_by_pos} and/or
+#' # \code{full_by_year} to maintain the full version of the specified labels.
+#'
+#' ggplot(df2,
+#'        aes(x = year2, y = ridership, color = system)) +
+#'   geom_line() +
+#'   scale_x_date(labels = abbr_years_date(full_by_year = c(2000)))
+#'
+#' # You can also remove the default maintenance of the first label and only
+#' # specify specific years.
+#' ggplot(df2,
+#'        aes(x = year, y = ridership, color = system)) +
+#'   geom_line() +
+#'   scale_x_continuous(labels = abbr_years_cont(full_by_pos = NULL,
+#'                                               full_by_year = c(1990,2020)))
+#'}
+#'
+#'@param full_by_pos Vector of integers, the position of breaks that should not
+#'  be abbreviated. This defaults to \code{c(1)}, which retains the original
+#'  first label and abbreviates subsequent ones. If all breaks should be
+#'  abbreviated, this can be set to NULL.
+#'@param full_by_year Vector of integers, the value of breaks that should not be
+#'  abbreviated. Defaults to NULL.
+#'
+#'@describeIn abbr_years_cont For continuous scales
+#'@export
+abbr_years_cont <- function(full_by_pos = c(1),
+                            full_by_year = NULL) {
+
+  fxn <- function(breaks) {
+
+    # Determine length of each break. Note that when breaks have been modified,
+    # ggplot sometimes adds an additional NA break in the first and last
+    # positions. We check whether this is the case and, if so, remove those for
+    # testing purposes.
+    lengths <- stringr::str_length(as.integer(breaks[which(!is.na(breaks))]))
+
+    # # Stop if the breaks are not in a four-digit format.
+    # if (!(min(lengths == 4) & max(lengths == 4))) {
+    #   stop("Breaks are not in a four-digit format. Cannot abbreviate.")
+    # }
+
+    # # Stop if breaks cannot be coerced to a number
+    # tryCatch({as.integer(breaks)},
+    #          warning = function(w){stop("Breaks cannot be coerced to a year. Cannot abbreviate.")})
```

Comment:
Why is the above code commented out? It seems like, if it is unnecessary, the lengths calculation in line 68 is also unnecessary?

Reply:
Oh, it's commented because I had this in an earlier iteration and it was causing it to break. I'll have to adjust all of these anyway when consolidating the two.